### PR TITLE
Avoid limit_current service call passing None-values and clarify error messages

### LINKED ISSUE #283: Error when calling limit_current: '<=' not supported between instances of 'int' and 'NoneType' after recent update
After updating to the latest version of the Zaptec integration, the service call zaptec.limit_current fails both in Node-RED and directly from Home Assistant Developer Tools.

Actual behavior:
The service fails with the following error: HomeAssistantError: Limit current failed: '<=' not supported between instances of 'int' and 'NoneType'

Additional information:
This setup has worked reliably for over two years.
The error occurs even when calling the service directly from Developer Tools, without Node-RED.
In Developer Tools → States, the charger entity no longer shows a numeric max_current attribute — it appears to be null.
It seems this attribute might have been removed in the latest version, but the limit_current logic still expects it to be a number.

Home Assistant core version: 2025.8.1

Zaptec integration version: 0.8.1

### PR DESCRIPTION
**Pull request by steinmn**
Fixes #283

### CODE REVIEW

File: custom_components/zaptec/services.py [hunk 1]
```diff
@@ -279,19 +279,20 @@ async def service_handle_upgrade_firmware(service_call: ServiceCall) -> None:
...
+        limit_args = {}
+        # only add the relevant arguments if they are not None
+        if available_current := service_call.data.get("available_current"):
+            limit_args["availableCurrent"] = available_current
+        if available_current_phase1 := service_call.data.get("available_current_phase1"):
+            limit_args["availableCurrentPhase1"] = available_current_phase1
+        if available_current_phase2 := service_call.data.get("available_current_phase2"):
+            limit_args["availableCurrentPhase2"] = available_current_phase2
+        if available_current_phase3 := service_call.data.get("available_current_phase3"):
+            limit_args["availableCurrentPhase3"] = available_current_phase3
```
Comment (sveinse) [inline custom_components/zaptec/services.py]: This won't work if the value is being set to 0. It needs to be explicitly tested for `None`.

File: custom_components/zaptec/api.py [hunk 2]
```diff
@@ -541,19 +541,26 @@ async def set_limit_current(self, **kwargs):
...
                 "availableCurrentPhase2",
                 "availableCurrentPhase3",
             )
-        )
-        if not (has_availablecurrent ^ has_availablecurrentphases):
+        ]
+        if not (has_availablecurrent ^ all(has_availablecurrentphases)):
             raise ValueError(
-                "Either availableCurrent or all of availableCurrentPhase1, "
+                "Either availableCurrent OR all of availableCurrentPhase1, "
```
Comment (sveinse) [inline custom_components/zaptec/api.py]: I'm not overly fond of using capital letters in the middle of an English sentence

File: custom_components/zaptec/api.py [hunk 3]
```diff
@@ -541,19 +541,26 @@ async def set_limit_current(self, **kwargs):
...
                 "availableCurrentPhase2, availableCurrentPhase3 must be set"
             )
+        if has_availablecurrent and any(has_availablecurrentphases):
+            raise ValueError(
+                "A combination of both availableCurrent and any of "
+                "availableCurrentPhase1, availableCurrentPhase2, "
+                "availableCurrentPhase3 is not allowed, choose one"
+            )
```
Comment (sveinse) [inline custom_components/zaptec/api.py]: This error is a little confusing to me. Perhaps it's better to test that if any of the three (with `any()`) have used, and all of them must be present. Error: "All of availableCurrent1, availableCurrent2 and availableCurrent2 must be present".

File: custom_components/zaptec/services.py (hunk 1 above)
Comment (steinmn) [reply custom_components/zaptec/services.py]: :facepalm: tried to be a little bit too elegant, fixed now

File: custom_components/zaptec/api.py (hunk 2 above)
Comment (steinmn) [reply custom_components/zaptec/api.py]: Wanted to add emphasis to make it clearer that it's an exclusive or, but I'm fine either way

File: custom_components/zaptec/api.py (hunk 3 above)
Comment (steinmn) [reply custom_components/zaptec/api.py]: Modified
